asreview/io/ris_reader.py

```diff
@@ -124,20 +122,36 @@ def read_data(cls, fp):
         ValueError
             File with unrecognized encoding is used as input.
         """
-        encodings = ['utf-8', 'utf-8-sig', 'ISO-8859-1']
+        encodings = ["utf-8", "utf-8-sig", "ISO-8859-1"]
         entries = None
-        for encoding in encodings:
-            try:
-                with open(fp, 'r', encoding=encoding) as bibliography_file:
-                    entries = list(
-                        rispy.load(bibliography_file, skip_unknown_tags=True))
-                    break
-            except UnicodeDecodeError:
-                pass
-            except IOError as e:
-                logging.warning(e)
         if entries is None:
-            raise ValueError("Cannot find proper encoding for data file.")
+            for encoding in encodings:
+                if is_url(fp):
+                    try:
+                        url_input = urlopen(fp)
+                        bibliography_file = io.StringIO(
+                            url_input.read().decode(encoding)
+                        )
+
+                        entries = list(
+                            rispy.load(bibliography_file, skip_unknown_tags=True)
+                        )
+                        bibliography_file.close()
+                    except UnicodeDecodeError:
+                        pass
```

Comment:
It is more efficient to iter the encodings after you downloaded the file. However, encoding detection might not be efficient at all and can be improved. Maybe even upstream. Just my brainwave.